File: boomerangPDS (copy)/src/main/java/boomerang/solver/MethodBasedFieldTransitionListener.scala
/**
 * ***************************************************************************** Copyright (c) 2018
 * Fraunhofer IEM, Paderborn, Germany. This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0.
 *
 * <p>SPDX-License-Identifier: EPL-2.0
 *
 * <p>Contributors: Johannes Spaeth - initial API and implementation
 * *****************************************************************************
 */
package boomerang.solver

import boomerang.scene.ControlFlowGraph
import boomerang.scene.Field
import boomerang.scene.Method
import boomerang.scene.Val
import sync.pds.solver.nodes.INode
import sync.pds.solver.nodes.Node
import wpds.impl.Transition
import wpds.impl.Weight
import wpds.impl.WeightedPAutomaton
import wpds.interfaces.WPAUpdateListener

abstract class MethodBasedFieldTransitionListener[W <: Weight](private val method: Nothing) extends Nothing {
  def getMethod: Nothing = method

  @Override def onWeightAdded(t: Nothing, w: W, aut: Nothing): Unit = {
    onAddedTransition(t)
  }

  def onAddedTransition(t: Nothing): Unit
}

File: boomerangPDS/boomerangPDS-2.5.1-sources/boomerang/solver/MethodBasedFieldTransitionListener.java
/*******************************************************************************
 * Copyright (c) 2018 Fraunhofer IEM, Paderborn, Germany.
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0.
 *  
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Johannes Spaeth - initial API and implementation
 *******************************************************************************/
package boomerang.solver;

import boomerang.jimple.Field;
import boomerang.jimple.Statement;
import boomerang.jimple.Val;
import soot.SootMethod;
import sync.pds.solver.nodes.INode;
import sync.pds.solver.nodes.Node;
import wpds.impl.Transition;
import wpds.impl.Weight;
import wpds.impl.WeightedPAutomaton;
import wpds.interfaces.WPAUpdateListener;

public abstract class MethodBasedFieldTransitionListener<W extends Weight>
        implements WPAUpdateListener<Field, INode<Node<Statement, Val>>, W> {
    private final SootMethod method;

    public MethodBasedFieldTransitionListener(SootMethod method) {
        this.method = method;
    }

    public SootMethod getMethod() {
        return method;
    }

    @Override
    public void onWeightAdded(Transition<Field, INode<Node<Statement, Val>>> t, W w,
            WeightedPAutomaton<Field, INode<Node<Statement, Val>>, W> aut) {
        onAddedTransition(t);
    }

    public abstract void onAddedTransition(Transition<Field, INode<Node<Statement, Val>>> t);
}
